latex_access/polish_braille.py

```diff
@@ -0,0 +1,391 @@
+        for letter in range (97,123): # Ascii, lower case
+            new_table["%c" % (letter)] = self.lowerLetter
+
+        self.table.update(new_table)
+
+    sqrt_with_two_args =re.compile(r".*\\sqrt\[(.*)\]+.*")
+
+    def before (self):
+        """Method ran before the translator at depth = 0.
+
+        Place anything here to do before we begin translation to polish braille. Eg. set
+        variables etc. Should NOT alter the actual input at all."""
+        self.lastnumber = -1 # keep track of where the last latex number # was open i.e. after a digit . or , (for Polish braille)
+    
+    def super(self,input,start,rting=()):
+        '''Translate  superscripts into polish braille.
+
+        Returns a touple with translated string and index of
+        first char after end of super.'''
+        arg=get_arg(input,start)
+		#Handle squared, degrees  and cubed as special cases
+        if arg[0] == "2":
+            translation=u"⠬⠆"
+        elif arg[0]=="3":
+            translation=u"⠬⠒"
+        elif arg[0]=="\circ":
+            translation=u"⠴"
+		#Handle primes
+        elif latex_access.primes.match(arg[0]):
+            translation=u"⠔"*arg[0].count("\\prime")
+        else:
+            if rting==():
+                translation=u"ó"
+                if arg[0].isdigit():
+                    for k in range(len(arg[0])) : translation+=self.table["L"+arg[0][k]]
+                elif arg[0][0] == "-" and arg[0][1].isdigit(): 
+                    translation+= u"-"
+                    for k in range(1,len(arg[0])) : translation+=self.table["L"+arg[0][k]]
+                else:
+                    translation+= self.translate(arg[0]) 
+            else:
+                translation=u"ó"
+                if arg[0][0] != "-" and arg[0][1].isdigit():
+                    for k in range(len(arg[0])) : translation+=self.table["L"+arg[0][k]]
+                elif arg[0][0] == "-" and arg[0][1].isdigit(): 
+                    translation+= u"-"
+                    for k in range(1,len(arg[0])) : translation+=self.table["L"+arg[0][k]]
+                elif arg[0][0] == "-" and not (arg[0][1].isdigit()) and arg[0][1] !="\\":
+                    translation+= u"-"
+                    translation+= u"⠠"
+                    translation+=  arg[0][1:]
+                elif arg[0][0] == "-" and arg[0][1] =="\\":
+                    translation+= u"-"
+                    translation+=  arg[0][1:]
+                elif arg[0][0] != "-" and arg[0][0] !="\\": 
+                    translation+= u"⠠"
+                    translation+=  arg[0]
+                elif arg[0][0] != "-" and arg[0][0] =="\\":
+                    translation+= arg[0]
+                translation = self.translate(translation,(rting[0]+arg[2],rting[1]+1))
+        return (translation,arg[1])
+
+
+    def sub(self,input,start,rting=()):
+        '''Translates subscripts to Polish braille.
+
+        Returns a touple, as above'''
```

Comment:
While this is a copy / paste relict please correct 'touple' to tuple in both translators - that should be pretty quick with find and replace.